Posts/2021/03/File-System-Watcher-Engine-Event.md

```diff
@@ -1,42 +1,54 @@
+---
+post_title: A Reusable File System Event Watcher for PowerShell
+username: msn
+categories: PowerShell
+tags: PowerShell, FileSystemWatcher, Module
+featured_image: 
+summary: FSWatcherEngineEvent module provides events of file system changes as powershell engine event
+---
+
 # A Reusable File System Event Watcher for PowerShell
 
-Have you ever needed a FilesystemWatcher and found it a bit cumbersome to interact with the
-[.Net frameworks FileSystemWatcher class](/dotnet/api/system.io.filesystemwatcher)
-within a powershell script?
+Some time ago I wanted to sync files from a source directory to a destination directory immediately after that had changed in the source directory.
+As a C# developer I'm aware of a .Net Framework class named '[FileSystemWatcher](/dotnetapi/system.io.filesystemwatcher)' which suits this job perfectly.
+A file system watcher listens to change notifications generated by the operating system and invokes a given function if the file change matches several filter criteria like the directory, the file name or the type of the change.
+There are already many examples in the internet how to create and configure the watcher in PowerShell but this isn't something I write down in 30 seconds from your memory in the moment I need it.
+I made the FSWatcherEngineEvent PowerShell module to make these file system watchers easier to use.
+It hides the C#-API behind a PowerShell command with argument completion, it keeps track of the created watchers and provides commands to pause notifications and to clean up the watchers if they are no longer needed.
+
+After you install and import the module, you can create a new filesystem watcher.
+As an example it will watch for changes in directory 'C:\Temp\files'.
```

Comment:
As an example, you can watch for changes in directory 'C:\Temp\files'.

Reply:
fixed